src/Parser/Druid/Restoration/Modules/Talents/Photosynthesis.js

```diff
@@ -49,17 +70,80 @@ class Photosynthesis extends Analyzer {
     const spellId = event.ability.guid;
     const amount = event.amount + (event.absorbed || 0);
 
-    if(spellId === SPELLS.REJUVENATION.id && this.selectedCombatant.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id, null, 0, 0, this.selectedCombatant.sourceID)) {
-      this.rejuvenationIncrease += calculateEffectiveHealing(event, PHOTOSYNTHESIS_REJUV_INCREASE);
+    // Lifebloom random bloom procc
+    if(spellId === SPELLS.LIFEBLOOM_BLOOM_HEAL.id && (this.lastRealBloomTimestamp === null || (event.timestamp - this.lastRealBloomTimestamp) < BLOOM_BUFFER_MS)){
+      this.lifebloomIncrease += amount;
     }
 
-    if(spellId === SPELLS.LIFEBLOOM_BLOOM_HEAL.id && (this.lastRealBloomTimestamp === null || (event.timestamp - this.lastRealBloomTimestamp) > BLOOM_BUFFER_MS)){
-      this.lifebloomIncrease += amount;
+    // Yes it actually buffs efflorescence, confirmed by Voulk and Bastas
+    if(this.selectedCombatant.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id) && (HOTS_AFFECTED_BY_ESSENCE_OF_GHANIR.includes(spellId) || spellId === SPELLS.EFFLORESCENCE_HEAL.id || spellId === SPELLS.SPRING_BLOSSOMS.id)) {
```

Comment:
For Photosynthesis's tick rate increase to be active it needs to be the selectedCombatant's own Lifebloom active on themselves. You can pass a SourceID to `hasBuff` to check for that.

Reply:
You're right